scylla-cdc/src/log_reader.rs

```diff
@@ -184,10 +185,10 @@ impl CDCReaderWorker {
                     stream_reader_tasks = self
                         .readers
                         .iter()
-                        .map(|reader| {
+                        .cartesian_product(self.table_names.clone())
+                        .map(|(reader, table_name)| {
                             let reader = Arc::clone(reader);
                             let keyspace = self.keyspace.clone();
-                            let table_name = self.table_name.clone();
                             let factory = Arc::clone(&self.consumer_factory);
                             tokio::spawn(async move {
                                 let consumer = factory.new_consumer().await;
```

Comment:
The consumer should be made aware from which table it consumes data. The new_consumer() method should accept the table/keyspace name.

Reply:
Are you sure you meant the source table? It's not a bad idea but I don't see how it's relevant here.
Maybe you meant the destination table? That would definitely be useful. But when I look at the traits we've defined
```rust
#[async_trait]
pub trait Consumer: Send {
    async fn consume_cdc(&mut self, data: CDCRow<'_>) -> anyhow::Result<()>;
}

#[async_trait]
pub trait ConsumerFactory: Sync + Send {
    async fn new_consumer(&self) -> Box<dyn Consumer>;
}
```

I don't see how a method like
```rust
async fn new_consumer(&self, dest_keyspace: &str, dest_table: &str) -> Box<dyn Consumer>
```
would fit such a general interface. For example `PrinterConsumer` doesn't have a need for destination table/keyspace.

Maybe we could have multiple `ReplicatorConsumerFactories`?

Reply:
I've just pushed the implementation with multiple factories for inspection.

Reply:
Sorry for the late response.

The more I think about the multi-table support the less sense I think it makes. I'm not sure there will be much benefit at the moment in supporting it natively in the CDCLogReader vs. just running multiple instances of the CDCLogReader (I just noticed that the scylla-cdc-java replicator does that).

Although running multiple CDCLogReaders duplicates work related to generation fetching, it is a tiny amount of work and complicating the CDCLogReader in order to save work isn't worth it IMO.

Let's drop the multi-table support and just change the replicator to run multiple instances of the CDCLogReader.

Reply:
A question then:

How many tables do we expect? Can the handles from `CDCLogReader`s be awaited with something like `join_all` or do we need use `FuturesUnordered` again?

Reply:
I don't think the number of tables will be big, but it is a user-controlled parameter so `FuturesUnordered` won't hurt here.